src/components/MapView/Download/index.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  Button,
+  createStyles,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Grid,
+  Hidden,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  Theme,
+  Typography,
+  WithStyles,
+  withStyles,
+} from '@material-ui/core';
+import Menu, { MenuProps } from '@material-ui/core/Menu';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faFileExport,
+  faCaretDown,
+  faImage,
+} from '@fortawesome/free-solid-svg-icons';
+
+import { jsPDF } from 'jspdf';
+
+import { useSelector } from 'react-redux';
+import { mapSelector } from '../../../context/mapStateSlice/selectors';
+
+const ExportMenu = withStyles((theme: Theme) => ({
+  paper: {
+    border: '1px solid #d3d4d5',
+    backgroundColor: theme.palette.primary.main,
+  },
+}))((props: MenuProps) => (
+  <Menu
+    elevation={0}
+    getContentAnchorEl={null}
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'center',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'center',
+    }}
+    {...props}
+  />
+));
+
+const ExportMenuItem = withStyles((theme: Theme) => ({
+  root: {
+    color: theme.palette.common.white,
+  },
+}))(MenuItem);
+
+function Download({ classes }: DownloadProps) {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [open, setOpen] = useState(false);
+  const selectedMap = useSelector(mapSelector);
+  const previewRef = useRef<HTMLCanvasElement>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  useEffect(() => {
+    if (open && selectedMap) {
+      const activeLayers = selectedMap.getCanvas();
+      const canvas = previewRef.current;
+      canvas!.setAttribute('width', activeLayers.width.toString());
+      canvas!.setAttribute('height', activeLayers.height.toString());
+      const context = canvas!.getContext('2d');
+      context!.drawImage(activeLayers, 0, 0);
+    }
+  });
+
+  const modalClose = () => {
+    setOpen(false);
+  };
+
+  const download = (format: String) => {
+    const ext = format === 'pdf' ? 'png' : format;
+    const canvas = previewRef!.current;
```

Comment:
Try avoiding the use of `canvas! ` and `previewRef!.` because we cannot guarantee these are not null, and its better to be safe with optional chaining `previewRef?.current`